internal/transport/transport.go

```diff
@@ -514,6 +501,35 @@ func Truncate(ctx context.Context, cfg *Config) error {
 	return nil
 }
 
+// Truncate will truncate the defined tables in the configuration.
+func Truncate(ctx context.Context, cfg *Config) error {
+	if !cfg.Truncate {
```

Comment:
My only note here is we can limit the number of times we call the `truncate` method by doing something like this:

```go
truncateRequest = new(proto.TruncateRequest)
if cfgTruncate {
    // Fill it with the root specific case
} else {
    // Fill it with the request specific case
}

return truncate(ctx, cfg, truncateRequest)
```

Reply:
lovely..